packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -376,6 +388,77 @@ contract L2Genesis is Script, Artifacts {
+        // deploy config, should we just not include it?
+        json = stdJson.serialize(serializedConfigName, "DAOForkSupport", false);
+        json = stdJson.serialize(serializedConfigName, "eip150Block", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "eip155Block", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "eip158Block", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "byzantiumBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "constantinopleBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "petersburgBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "istanbulBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "muirGlacierBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "berlinBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "londonBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "arrowGlacierBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "grayGlacierBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "mergeNetsplitBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "terminalTotalDifficulty", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "terminalTotalDifficultyPassed", true);
+        // TODO This was previously set as uint64, is it okay to be uint256?
+        json = stdJson.serialize(serializedConfigName, "bedrockBlock", cfg.l2GenesisBlockNumber());
+        json = stdJson.serialize(serializedConfigName, "regolithTime", _getHardforkTime(cfg.l2GenesisRegolithTimeOffset(), genesisTime));
+        json = stdJson.serialize(serializedConfigName, "canyonTime", _getHardforkTime(cfg.l2GenesisCanyonTimeOffset(), genesisTime));
+
+        // TODO
+        json = stdJson.serialize(serializedConfigName, "shanghaiTime", uint256(0));
+        // TODO
+        json = stdJson.serialize(serializedConfigName, "cancunTime", uint256(0));
+        // TODO
+        json = stdJson.serialize(serializedConfigName, "ecotoneTime", uint256(0));
+        // TODO
+        json = stdJson.serialize(serializedConfigName, "interopTime", uint256(0));
+
+        string memory serializedOptimismConfig = "optimismConfig";
+        string memory nestedOptimismConfig = "";
+        // TODO
+        nestedOptimismConfig = stdJson.serialize(serializedOptimismConfig, "eip1559Elasticity", uint256(6));
+        // TODO
+        nestedOptimismConfig = stdJson.serialize(serializedOptimismConfig, "eip1559Denominator", uint256(50));
+        // TODO
+        nestedOptimismConfig = stdJson.serialize(serializedOptimismConfig, "eip1559DenominatorCanyon", uint256(250));
+
+        json = stdJson.serialize(serializedConfigName, "optimism", nestedOptimismConfig);
+        return json;
+    }
```

Comment:
@tynes Still WIP, but is this how we should add these properties? The following are additional properties that aren't being added yet, but I'm not sure if every property needs to be:

```json
"nonce": "0x0",
  "timestamp": "0x65bc26af",
  "extraData": "0x424544524f434b",
  "gasLimit": "0x1c9c380",
  "difficulty": "0x0",
  "mixHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "coinbase": "0x4200000000000000000000000000000000000011",
"number": "0x0",
  "gasUsed": "0x0",
  "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "baseFeePerGas": "0x1",
  "excessBlobGas": null,
  "blobGasUsed": null
```

Reply:
we need coinbase for sure. and also gas limit. we should use the extradata which says `BEDROCK` in hex. it was previously configurable, could do a fallback to default value of bedrock but it simplifies things if its always BEDROCK in the genesis

Reply:
Off of the top of my head i am not sure which exactly are needed and which are not. I think if we look at the `Genesis` type in geth, in the `UnmarshalJSON` function, it should have default empty values if they are not set

Reply:
at the end of the day we might need to update something if it doesn't work in the end to end devnet setup but that is ok

Reply:
Judging from [this](https://github.com/ethereum-optimism/op-geth/blob/b6d07325ba02be62197fa0cbab347d6be9e8cf8f/core/genesis.go#L53-L61), it looks like they are all required except:

```
"excessBlobGas": null,
"blobGasUsed": null
```

I've elected not to include these since [is says not to](https://github.com/ethereum-optimism/op-geth/blob/b6d07325ba02be62197fa0cbab347d6be9e8cf8f/core/genesis.go#L63-L64):

```
// These fields are used for consensus tests. Please don't use them
// in actual genesis blocks.
```